### PR TITLE
[P4 GSoC 2024] Set up final report directory

### PR DESCRIPTION
**Pull request by qobilidop**
Also updated repo README with more project details.

### CODE REVIEW

**Review by Dscano — APPROVED** [review]
Looks good to me

**Review by fruffy — APPROVED** [review]

File: 2024/report/AdarshRawat1/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# P4 Compiler Documentation
```
Comment (fruffy) [inline 2024/report/AdarshRawat1/README.md:1]: Maybe make the project folder structure about the project, not the user name? 

E.g.
```
- documentation
- pna_bmv2
- formatter
- container_migration
```

Comment (rst0git) [reply 2024/report/AdarshRawat1/README.md:1]: In addition to Fabian's suggestion, it might be better to rename the folder `report` to `projects`.

Comment (qobilidop) [reply 2024/report/AdarshRawat1/README.md]: Done.

**Pull request merged**